__tests__/app.test.js

```diff
@@ -164,5 +159,74 @@ describe("post/api/articles/:articleid/comments", () => {
       });
   });
 });
+test("returns 404 error if article id not a number", () => {
+  const comment = {
+    username: "icellusedkars",
+    body: "Great article!",
+  };
+  return request(app)
+    .post("/api/articles/alex/comments")
+    .send(comment)
+    .expect(404)
+    .then(({ body }) => {
+      expect(body).toEqual({ msg: "404, article not found" });
+    });
+});
 
+test("returns 404 error if no comment given", () => {
+  const comment = {
+    username: "icellusedkars",
+  };
+  return request(app)
+    .post("/api/articles/alex/comments")
+    .send(comment)
+    .expect(404)
+    .then(({ body }) => {
+      expect(body).toEqual({ msg: "404, article not found" });
+    });
+});
 
+describe("patch/api/articles/:article_id", () => {
+  test("updates the votes of an article by given paramater, no existing votes", () => {
+    const votes = { inc_votes: 10 };
+    return request(app)
+      .patch("/api/articles/3")
+      .send(votes)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(
+          expect.objectContaining({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            article_img_url: expect.any(String),
+            votes: expect.any(Number),
+            author: expect.any(String),
+            body: expect.any(String),
+          })
+        );
+        expect(body.votes).toBe(10);
+      });
+  });
+  test("returns 404 error if invalid article", () => {
+    const votes = { inc_votes: 10 };
+    return request(app)
+      .patch("/api/articles/5000")
+      .send(votes)
+      .expect(404)
+      .then((body) => {
+        expect(body._body.msg).toBe("Invalid article ID or no votes passed");
```

Comment:
This error message is too vague - it covers two entirely different reasons. Try to think how you could first check if the article exists and throw an error if it doesn't. Otherwise, think of a way that you can check whether the request has a votes key - again, if it doesn't, make sure this is expressed in it's own error.

Reply:
What about an article id of the wrong data type?